### PR TITLE
#CU-31wbqcz #xcmp #statemine and #karura attempts to specify flows

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Issue

It is hard to encode all checks which should be done to send proper channel message.

This another attempt to document it.

## Additional Changes

- statemine as it was backed to democracy
- karura as it was backed to democracy
- guide which was followed to check each step in message
- traps guide
- allow root calls on dali 

## Checklist

- Links should open some valid locations.

### CODE REVIEW

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 1]
```diff
@@ -1,53 +1,47 @@
...
-4. Make USDT priceable on Picasso
-
-All steps are SCALE encoded and can be decoded and executed by named consensuses.
-
-Proposal follows same steps as other chains did.
 
 ## Preparation
 
-One should have `Identity` on Kusama to create `Proposal` on  https://parachains.polkassembly.io/ . 
+One should have `Identity` on Kusama to create `Proposal` on   .
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply rfcs/0003-xcmp-picasso-statemine.md]: fixed

**Review by PoisonPhang — APPROVED** [review]

File: docs/docs/internal/xcmp/how-to-open-xcmp-channel.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+[0x3c00e8030000e803000000900100](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/extrinsics/decode/0x3c00e8030000e803000000900100).
+
+Check `Transact` message has proper existing parachain recipient.
+
+`proposedMaxCapacity` is less or equal to `Storage -> configuration -> activeConfig` `hrmpChannelMaxCapacity`
+
+`proposedMaxMessageSize` is less or equal to `hrmpChannelMaxMessageSize` here.
+
+`hrmpMaxParachain*Channels` is not exceeded by neither participant chains.
+
```
Comment (Martin-Kiss) [inline docs/docs/internal/xcmp/how-to-open-xcmp-channel.md:80]:
```suggestion
Check that: 

- The `Transact` message contains a proper existing parachain recipient.

- `proposedMaxCapacity` is less or equal to `Storage -> configuration -> activeConfig` `hrmpChannelMaxCapacity`

- `proposedMaxMessageSize` is less or equal to `hrmpChannelMaxMessageSize` here.

- `hrmpMaxParachain*Channels` is not exceeded by any participant chains.
```
Format

File: docs/docs/internal/xcmp/how-to-open-xcmp-channel.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+## Counterparty chain to accept our request and request back
+
+Encode above preimage via `Democracy -> Submit preimage`.
+
+Then `Submit proposal`.
```
Comment (Martin-Kiss) [inline docs/docs/internal/xcmp/how-to-open-xcmp-channel.md:85]:
```suggestion
## Counterparty chain to accept our request and request back

Encode the preimage above via `Democracy -> Submit preimage`.

Then `Submit proposal`.
```

File: docs/docs/internal/xcmp/how-to-open-xcmp-channel.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkarura-rpc-3.aca-api.network%2Fws#/extrinsics/decode/0x33000101000210000400000000070010a5d4e81300000000070010a5d4e800060003009435775c1802083c01270800003c0027080000e8030000009001000d010004000100411f
+
+**Details** 
+
+Find latest `proposal` or success `referenda` to open channels and compare difference with our message. Things change, so further example away more likely it is not up to date.
+
+Please do all relevant checks from previous step.
```
Comment (Martin-Kiss) [inline docs/docs/internal/xcmp/how-to-open-xcmp-channel.md:93]:
```suggestion
**Details** 

Find the latest `proposal` or successful `referenda` to open channels and compare differences with our message. Things change so the further examples are likely to be out of date.

Please do all relevant checks from the previous step.
```
Ambiguous; Is the assumption in my suggestion correct or could you elaborate on the examples being out of date?

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 5]
```diff
@@ -1,277 +1,32 @@
...
+```markdown
+Picasso aims to become the DeFi hub of Kusama, connecting parachains with the broader Cosmos ecosystem. Through opening this XCM channel, Statemine assets will have access to Picasso dapps, thereby increasing the utility and TVL of USDT and other Statemine tokens.
 
+For example, we will setup pools for USDT and KSM on Pablo, as well as integrating them within our later released pallets such as CosmWasm. This opens up the possibility to also support these tokens within our IBC bridge that will launch in the future too. 
 
-## References
+### Flow
 
-- https://kusama.polkassembly.io/referendum/163
+- A force transfer 11 KSM from Kusama treasury (`F3opxRbN5ZbjJNU511Kj2TLuzFcDq9BGduA9TgiECafpg29`) to Statemine (`F7fq1jSNVTPfJmaHaXCMtatT1EZefCUsa7rRiQVNR5efcah`). 10 KSM used as deposit for channels. 1 KSM for transaction fees payments.
```
Comment (Martin-Kiss) [inline rfcs/0003-xcmp-picasso-statemine.md:19]:
```suggestion
- In a forced transfer of 11 KSM from Kusama treasury (`F3opxRbN5ZbjJNU511Kj2TLuzFcDq9BGduA9TgiECafpg29`) to Statemine (`F7fq1jSNVTPfJmaHaXCMtatT1EZefCUsa7rRiQVNR5efcah`). 10 KSM are used as a deposit for channels. 1 KSM is used to pay transaction fees.
```

File: rfcs/0004-xcmp-picasso-karura.md [hunk 6]
```diff
@@ -1,167 +1,37 @@
...
-<!-- cspell:disable-next -->
-Parachain is `5Ec4AhNu5P23CC2GdrmsC8HYBKiidqhPnfSfRQPaH9swr2b7` and make sure it has 20 ROC.
+- Accepts existing open request channel from Picasso
+- Sends open request to Picasso
 
-Transfer to [dali-rococo] accounts some amounts.
+### Requires
 
-## Picasso to Karura
+12 KSM to be on Parachain 2000(Karura) account
```
Comment (Martin-Kiss) [inline rfcs/0004-xcmp-picasso-karura.md:25]:
```suggestion
This requires 12 KSM to be on the target Parachain 2000(Karura) account.
```
Nit format (see edit line 23)

File: rfcs/0004-xcmp-picasso-karura.md [hunk 7]
```diff
@@ -1,167 +1,37 @@
...
-Root is `5D2cjLGNWibiSsEh4oPXo8MmBTkeptrpuiUdEKRMauUn1TkZ` and make sure it has 20 ROC.
+### Flow
 
-<!-- cspell:disable-next -->
-Parachain is `5Ec4AhNu5P23CC2GdrmsC8HYBKiidqhPnfSfRQPaH9swr2b7` and make sure it has 20 ROC.
+- Accepts existing open request channel from Picasso
+- Sends open request to Picasso
 
-Transfer to [dali-rococo] accounts some amounts.
+### Requires
```
Comment (Martin-Kiss) [inline rfcs/0004-xcmp-picasso-karura.md:23]: Nit format (see edit line 25)

File: code/integration-tests/local-integration-tests/README.md [hunk 8]
```diff
@@ -1,31 +1,3 @@
...
-
-In each reserver transfer test ensure sibling accounts, para accounts are set up well and total supplys always same in the end.
-
-Remove all usage of XTokens as these violate XCM guidlines and do not work well.
-
-Ensure that assets from pallets encoded like (parent = 0, pallet = 44, index = 1232132).
-
-Run attacks which try to spend parachain account, spend other person amount.
\ No newline at end of file
+In memory tests running several parachains and relay chain.
```
Comment (Martin-Kiss) [inline code/integration-tests/local-integration-tests/README.md:3]: This seems unfinished.

**Review by Martin-Kiss — COMMENTED** [review]
A few questions for accuracy and some edits.

File: docs/docs/internal/xcmp/how-to-open-xcmp-channel.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
+# Overview 
+
+This document describes all technical steps required to open channel to arbitrary.
+
+Please adapt it to variation of governance, business value and discussion forums according target chains.
+
+In this guide there are some examples and links used with guiding explanation on how to modify this for next channel.
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply docs/docs/internal/xcmp/how-to-open-xcmp-channel.md]: XCMP channel (as in file name)

File: docs/docs/internal/xcmp/how-to-open-xcmp-channel.md (hunk 4 above)
Comment (dzmitry-lahoda) [reply docs/docs/internal/xcmp/how-to-open-xcmp-channel.md:93]: yes.

Comment (dzmitry-lahoda) [reply docs/docs/internal/xcmp/how-to-open-xcmp-channel.md:93]: i will remove this lines - just wanted to say there is much details and churn - so this guide is no 100% exact at time one may read it.

File: docs/docs/internal/xcmp/how-to-open-xcmp-channel.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
+# Overview 
+
+This document describes all technical steps required to open channel to arbitrary.
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply docs/docs/internal/xcmp/how-to-open-xcmp-channel.md:3]: yes

File: code/integration-tests/local-integration-tests/README.md (hunk 8 above)
Comment (dzmitry-lahoda) [reply code/integration-tests/local-integration-tests/README.md:3]: :) Finsihed, so will tune.

File: docs/docs/internal/xcmp/how-to-open-xcmp-channel.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+Please adapt it to variation of governance, business value and discussion forums according target chains.
+
+In this guide there are some examples and links used with guiding explanation on how to modify this for next channel.
+
+We are changing ecosystem, documents obsolete , like [moonbeam], and stop working.
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply docs/docs/internal/xcmp/how-to-open-xcmp-channel.md]: removed comment. changes in env can break docs - anyway expected. `enacting a specific change in our ecosystem that causes obsolete documents` - this.

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-31wbqcz Specify the test cases to be run](https://app.clickup.com/t/31wbqcz)

**Comment by dzmitry-lahoda** [comment]
@mergifyio refresh

**Comment by dzmitry-lahoda** [comment]
@Mergify merge

**Comment by dzmitry-lahoda** [comment]
Soon changes will be revereted https://composablefinance.slack.com/archives/C03LBRRMU4U/p1668506492266039?thread_ts=1668430464.509469&cid=C03LBRRMU4U, and likely merge conflict will vanish. So will be able to merge.

**Pull request merged**